### PR TITLE
Feature/get versions of attribute

### PR DESCRIPTION
**Pull request by Milena-Czierlinski**

### CODE REVIEW

**Comment by Milena-Czierlinski** [comment]
I plan to put the functionality of the GetSharedVersionsOfIdentityAttributeUseCase in the consumption lib in a next iteration, so that it is available for other usecases, too, e.g. ShareIdentityAttributeUseCase.

File: src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            return Result.ok(AttributeMapper.toAttributeDTOList(allVersions));
+        }
+
+        const query: GetAttributesRequestQuery = {};
+        query["shareInfo.sourceAttribute"] = allVersions.map((x) => x.id.toString());
+
+        const flattenedQuery = flattenObject(query);
+        const dbQuery = GetAttributesUseCase.queryTranslator.parse(flattenedQuery);
+
+        const sharedAttributes = await this.attributeController.getLocalAttributes(dbQuery);
```
Comment (stnmtz) [inline src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts:49]: We should not use the queryTranslator here, rather following syntax (as db queries):

```suggestion
        const query = {
                "shareInfo.sourceAttribute": { $in: allVersions.map((x) => x.id.toString())}
        };

        const sharedAttributes = await this.attributeController.getLocalAttributes(query);
```

File: src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        return Result.ok(AttributeMapper.toAttributeDTOList(sharedAttributes));
```
Comment (stnmtz) [inline src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts:51]: Hmm, we return the list unsorted - should we sort it manually?

**Review by stnmtz — COMMENTED** [review]
I wasn't able to look over the tests yet. Additionally, please "park" an idea, that we add an additional optional query parameter for both use-cases in order to enhance the query.

File: src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts (hunk 2 above)
Comment (Milena-Czierlinski) [reply src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts:47]: I plan to change the implementation, when it's moved to the consumption lib, anyways such that a sorted list is returned. Would you like me to think about a way to sort the result already here nevertheless?

Comment (stnmtz) [reply src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts:47]: Fine for me

**Review by stnmtz — APPROVED** [review]
It seems like you are not testing getSharedVersionsOfIdentityAttribute with multiple peers (having received an attribute, having receive an attribute succession) - please add this to our todo list.

**Pull request merged**